sstables_loader.cc

```diff
@@ -364,20 +365,17 @@ future<std::vector<tablet_sstable_collection>> get_sstables_for_tablets(const st
             const auto& sst = *sst_it;
             auto sst_first = sst->get_first_decorated_key().token();
             auto sst_last = sst->get_last_decorated_key().token();
-            auto sst_token_range = dht::token_range{sst_first, sst_last};
-
-            // sstables are sorted by first key, so should skip this SSTable since it
-            // doesn't overlap with the current tablet range.
-            if (!tablet_range.overlaps(sst_token_range, dht::token_comparator{})) {
-                // If the start of the next SSTable's token range lies beyond the current tablet's token
-                // range, we can safely conclude that no more relevant SSTables remain for this tablet.
-                if (tablet_range.after(sst_token_range.start()->value(), dht::token_comparator{})) {
-                    break;
-                }
+
+            // SSTable entirely after tablet -> no further SSTables (larger keys) can overlap
```

Comment:
The list of sstables that we walk here, is it sorted ascending or descenging? And IIRC the reverse view is (still) in use? I thought that plan was to improve the reverse-soring-vs-reverse-iterator mess :D Please, clarify

Reply:
unfortunately, at this point we have to keep it sorted backwards since in the `sstable_streamer::stream_sstables` we are chopping sstables from the end of the vector. We can dump this mess once we change the container type we use that works well with removing entities from the head